docs/spikes/github-actions-spike.md

```diff
@@ -0,0 +1,163 @@
+
+So we'd simply tag the container with a version number when pushing it up to a repository as as follows:
+  
+```
+- name: Build and push
+  uses: docker/build-push-action@v3
+  with:
+    file: ${{ inputs.container-path }}
+    context: ${{ inputs.container-context }}
+    push: true
+    tags: |
+      ghcr.io/organisation/container-name:latest
+      ghcr.io/organisation/container-name:1.0.${{github.run_number}} 
+```
+Breaking changes would require changes to the pipeline to increment the major or minor version number, or store that value in a variable.  
+
+If we wanted to version the assemblies, we could simply modify the application code in the pipeline to specify this value and _then_ build the code.
+
+## Workflows calling other workflows with parameters
+
+This can be done using [Composite Actions](https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/)
+
+Actions should be located in a `.github/actions` folder
+
+  ![GitHub Actions](../images/github_actions.png)
+
+Each action must:
+- Be in a named directory in the actions folder and contain one file called `action.yml`
+- Have a name and description
+- Inputs must be specified
+- Include the syntax
+    ```
+    runs: 
+      using: composite
+    ``` 
+- Secrets must be passed to composite actions as parameters (inputs)
+
+### Composite workflows in action
+**In the main workflow - workflow.yml**
+- Reference the action by the name of the folder it sits in
+```
+...
+ - name: Deploy container
+   uses: ./.github/actions/deploy-container
+   with:
+     ghcr-username: ${{ github.repository_owner }}
+...
+```
+**Basic action.yml template**
+```
+name: "Deploy container to ghcr"
+description: "Deploys container to github container registry"
+inputs:
+  ghcr-username:
+    requred: true
+    description: "Username for ghcr"
+  ghcr-password:
+    requried: true
+    description: "Password for ghcr"
+runs:
+  using: "composite"
+  steps:
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ inputs.ghcr-username }}
+        password: ${{ inputs.ghcr-password }}
+```
+
+An alternative method is using [Reusable Workflows](https://github.blog/2022-02-10-using-reusable-workflows-github-actions/) but these can be slightly more limiting
+
+
+## Uploading Artifacts
+
+Github Actions has the capability to store artifacts produced by the pipeline. 
+
+For this we can use the ```actions/upload-artifact``` default action.
+
+```
+...
+ - name: Upload Mutation Report
+   uses: actions/upload-artifact@v3.0.0
+   with:
+     name: artifact-name
+     path: (Directory or File)
+...
+```
+Point the ```path``` variable at a directory name or a file and the action will upload it into pipeline storage.
+
+Once the build is complete - the artefacts are visible in the pipeline summary:

```

Comment:
Though I prefer the British spelling for this word, let's keep it consistent with the rest of the doc
```suggestion
Once the build is complete - the artifacts are visible in the pipeline summary:

```